pkg/smb/nodeserver_test.go

```diff
@@ -61,52 +57,72 @@ func TestNodeStageVolume(t *testing.T) {
+			},
 		},
 		{
-			desc:        "[Error] Volume capabilities missing",
-			req:         csi.NodeStageVolumeRequest{VolumeId: "vol_1"},
-			expectedErr: status.Error(codes.InvalidArgument, "Volume capability not provided"),
+			desc: "[Error] Volume capabilities missing",
+			req:  csi.NodeStageVolumeRequest{VolumeId: "vol_1"},
+			expectedErr: testutil.TestError{
+				DefaultError: status.Error(codes.InvalidArgument, "Volume capability not provided"),
+			},
 		},
 		{
-			desc:        "[Error] Stage target path missing",
-			req:         csi.NodeStageVolumeRequest{VolumeId: "vol_1", VolumeCapability: &stdVolCap},
-			expectedErr: status.Error(codes.InvalidArgument, "Staging target not provided"),
+			desc: "[Error] Stage target path missing",
+			req:  csi.NodeStageVolumeRequest{VolumeId: "vol_1", VolumeCapability: &stdVolCap},
+			expectedErr: testutil.TestError{
+				DefaultError: status.Error(codes.InvalidArgument, "Staging target not provided"),
+			},
 		},
 		{
 			desc: "[Error] Source field is missing in context",
 			req: csi.NodeStageVolumeRequest{VolumeId: "vol_1", StagingTargetPath: sourceTest,
 				VolumeCapability: &stdVolCap},
-			expectedErr: status.Error(codes.InvalidArgument, "source field is missing, current context: map[]"),
+			expectedErr: testutil.TestError{
+				DefaultError: status.Error(codes.InvalidArgument, "source field is missing, current context: map[]"),
+			},
 		},
 		{
 			desc: "[Error] Not a Directory",
 			req: csi.NodeStageVolumeRequest{VolumeId: "vol_1##", StagingTargetPath: smbFile,
 				VolumeCapability: &stdVolCap,
 				VolumeContext:    volContext,
 				Secrets:          secrets},
-			expectedErr: status.Error(codes.Internal, "MkdirAll ./smb.go failed with error: mkdir ./smb.go: not a directory"),
+			expectedErr: testutil.TestError{
+				DefaultError: status.Error(codes.Internal, fmt.Sprintf("MkdirAll %s failed with error: mkdir %s: not a directory", smbFile, smbFile)),
+				WindowsError: status.Error(codes.Internal, fmt.Sprintf("Could not mount target %s: mkdir %s: The system cannot find the path specified.", smbFile, smbFile)),
+			},
 		},
 		{
 			desc: "[Error] Failed SMB mount mocked by MountSensitive",
 			req: csi.NodeStageVolumeRequest{VolumeId: "vol_1##", StagingTargetPath: errorMountSensSource,
 				VolumeCapability: &stdVolCap,
 				VolumeContext:    volContext,
 				Secrets:          secrets},
-			expectedErr: status.Errorf(codes.Internal, "volume(vol_1##) mount \"test_source\" on \"./error_mount_sens_source\" failed with fake MountSensitive: target error"),
+			expectedErr: testutil.TestError{
+				DefaultError: status.Errorf(codes.Internal,
+					fmt.Sprintf("volume(vol_1##) mount \"test_source\" on \"%s\" failed with fake MountSensitive: target error",
+						errorMountSensSource)),
+				// Not a desired error. This will need a better fix
+				WindowsError: fmt.Errorf("prepare stage path failed for %s with error: could not cast to csi proxy class", errorMountSensSource),
+			},
```

Comment:
This is not the desired error. While we would expect a target error, this test fails in Windows due to some type casting. We'll need a fix for the long term. I' not sure why this is happening. Any ideas?

Reply:
add a `// todo: `